File: 2866-longest-even-odd-subarray-with-threshold/2866-longest-even-odd-subarray-with-threshold.java
class Solution {
    public int longestAlternatingSubarray(int[] nums, int threshold) {
        int res = 0;
        int curr = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] % 2 == 0 && nums[i] <= threshold) {
                curr = 1;
                res = Math.max(res, curr);
                while (i + 1 < nums.length && nums[i] % 2 != nums[i + 1] % 2 && nums[i+1] <= threshold) {
                    curr++;
                    res = Math.max(res, curr);
                    i++;
                }
            } else {
                curr = 0;
            }
        }
        return res;
    }
}

File: 2866-longest-even-odd-subarray-with-threshold/2866-longest-even-odd-subarray-with-threshold.py
class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        res = 0
        curr = 0
        for i in range(len(nums)):
            if nums[i] % 2 == 0 and nums[i] <=threshold:
                curr = 1
                res = max(res , curr)
                while i + 1 < len(nums) and nums[i] % 2 != nums[i + 1] % 2 and nums[i+1] <=threshold:
                    curr += 1
                    res = max(res , curr)
                    i += 1
            else:
                curr = 0
        return res